docs/usage/dns-search-path-optimization.md

```diff
@@ -0,0 +1,120 @@
+qualified names. The DNS option `ndots` specifies how many dots (`.`) a name needs to have to be considered fully qualified.
+For names with less than `ndots` dots (`.`), the [DNS search path](#dns-search-path) will be applied.
+
+## DNS Search Path, `ndots` and Kubernetes
+
+Kubernetes tries to make it easy/convenient for developers to use name resolution. It provides several means to address a
+service, most notably by its name directly, using the namespace as suffix, utilizing `namespace.svc` as suffix or as a fully
+qualified name as `service-name.namespace-name.svc.cluster.local` (assuming `cluster.local` to be the cluster domain).
+
+This is why the DNS search path is fairly long in Kubernetes, usually consisting of `namespace-name.svc.cluster.local`,
+`svc.cluster.local`, `cluster.local` and potentially some additional entries coming from the local network of the cluster.
+For various reasons, the default `ndots` value in the context of Kubernetes is with `5` also fairly large. See
+[this comment](https://github.com/kubernetes/kubernetes/issues/33554#issuecomment-266251056) for a more detailed description.
+
+## DNS Search Path/`ndots` Problem in Kubernetes
+
+As the DNS search path is long and `ndots` is large, a lot of DNS queries might traverse the DNS search path. This results
+in an explosion of DNS requests.
+
+For example, consider the name resolution of the default kubernetes service `kubernetes.default.svc.cluster.local`. As this
+name has only four dots it is not considered a fully qualified name according to the default `ndots=5` setting. Therefore,
+the DNS search path is applied resulting in the following queries being created
+ - `kubernetes.default.svc.cluster.local.some-namespace.svc.cluster.local`
+ - `kubernetes.default.svc.cluster.local.svc.cluster.local`
+ - `kubernetes.default.svc.cluster.local.cluster.local`
+ - `kubernetes.default.svc.cluster.local.network-domain`
+ - ...
+
+In IPv4/IPv6 dual stack systems, the amount of DNS requests may even double as each name is resolved for IPv4 and IPv6.
+
+## General Workarounds/Mitigations
+
+Kubernetes provides the capability to set the DNS options for each pod (see
+[Pod DNS config](https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#pod-dns-config) for details).
+However, this has to be applied for every pod (doing name resolution) to resolve the problem. An mutating web hook may be
+useful in this regard. Unfortunately, the DNS requirements may be different depending on the workload. Therefore, a general
+solution may difficult to impossible.
+
+Another approach is to use always fully qualified names and append a dot (`.`) to the name to prevent the name resolution
+system from using the DNS search path. This might be somewhat counterintuitive as most developers are not used to the
+trailing dot (`.`). Furthermore, it makes moving to different landscapes more difficult/error-prone.
```

Comment:
If it is known that a name is fully qualified adding a dot is a simple approach indeed. The problem is that
1. It is not always clear whether a name is fully qualified or not, e.g. if a name is passed around or taken from a configuration file.
2. Nearly nobody knows/uses it like this.

I am not arguing that it is a bad approach. It solves the problem for fully qualified names in the right way.
That being said I am very doubtful if we can educate our users that this is way going forward. Apparently, the kubernetes community was not successful in that regard or otherwise we would have never seen these issues.
Futhermore, it might be very difficult to change larger applications/services consisting of multiple entities.